core/aws-core/src/main/java/software/amazon/awssdk/awscore/exception/AwsServiceException.java

```diff
@@ -62,10 +62,13 @@ public AwsErrorDetails awsErrorDetails() {
     @Override
     public String getMessage() {
         if (awsErrorDetails != null) {
+            String extendedRequestId =
+                awsErrorDetails().sdkHttpResponse().firstMatchingHeader("x-amz-id-2").orElse("not available");
             return awsErrorDetails().errorMessage() +
                     " (Service: " + awsErrorDetails().serviceName() +
                     ", Status Code: " + statusCode() +
-                    ", Request ID: " + requestId() + ")";
+                    ", Request ID: " + requestId() +
+                    ", Extended Request ID: " + extendedRequestId + ")";
```

Comment:
I realize it's actually DRYer to do this here, but for consistency can we set the extendedRequestId the same way we set the requestId: when the exception is created? (And maybe just print null instead of not available)

We can add a getter for retrieving it (that sometimes returns null).

Reply:
Yep, that's actually how I initially implemented this, but I was trying to make this simpler because I was worried the change would be too complicated.  I'll put out the original version today.